src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.ConvertToInteger.cs

```diff
@@ -32,23 +32,23 @@ public static void ConvertToInteger<TFrom, TTo>(ReadOnlySpan<TFrom> source, Span
             public static TTo Invoke(TFrom x) => TFrom.ConvertToInteger<TTo>(x);
 
             public static bool Vectorizable =>
-                (typeof(TFrom) == typeof(float) && typeof(TTo) == typeof(int)) ||
-                (typeof(TFrom) == typeof(float) && typeof(TTo) == typeof(uint)) ||
-                (typeof(TFrom) == typeof(double) && typeof(TTo) == typeof(long)) ||
-                (typeof(TFrom) == typeof(double) && typeof(TTo) == typeof(ulong));
+                (typeof(TFrom) == typeof(float) && IsInt32Like<TTo>()) ||
+                (typeof(TFrom) == typeof(float) && IsUInt32Like<TTo>()) ||
+                (typeof(TFrom) == typeof(double) && IsInt64Like<TTo>()) ||
+                (typeof(TFrom) == typeof(double) && IsUInt64Like<TTo>());
 
             public static Vector128<TTo> Invoke(Vector128<TFrom> x)
             {
                 if (typeof(TFrom) == typeof(float))
                 {
-                    if (typeof(TTo) == typeof(int)) return Vector128.ConvertToInt32(x.AsSingle()).As<int, TTo>();
-                    if (typeof(TTo) == typeof(uint)) return Vector128.ConvertToUInt32(x.AsSingle()).As<uint, TTo>();
+                    if (IsInt32Like<TTo>()) return Vector128.ConvertToInt32(x.AsSingle()).As<int, TTo>();
+                    if (IsUInt32Like<TTo>()) return Vector128.ConvertToUInt32(x.AsSingle()).As<uint, TTo>();
```

Comment:
Given then `Vectorizable` check, this could be (with optional asserts):
```csharp
if (TTo.IsNegative(TTo.MinValue))
{
    return Vector128.ConvertToInt32(x.AsSingle()).As<int, TTo>();
}
return Vector128.ConvertToUInt32(x.AsSingle()).As<uint, TTo>();
```

It leads to much smaller IL that the JIT can trivially see is constant, without depending on the inliner

Reply:
You could do `Vector128.IsNegative<TTo>(Vector128<T>.AllBitsSet) != Vector128<TTo>.Zero`. This works since we know `tto` is an integer and `AllBitsSet` will either be `-1` or `MaxValue`.

-- It's even simpler in .NET 10, but general intent is to try and avoid needing a second call to determine if a branch is dead or not, since that will impact inline profitability for these core helpers and we've already seen a few cases where the JIT gives up around that due to the size of the methods they're being inlined into.

Reply:
> You could do Vector128.IsNegative<TTo>(Vector128<T>.AllBitsSet) != Vector128<TTo>.Zero

And this is guaranteed to be constant folded down such that the expression and one or the other branches is eliminated?